src/or/hs_config.c

```diff
@@ -266,6 +265,94 @@ config_service_v3(const config_line_t *line_,
       have_num_ip = 1;
       continue;
     }
+    /* Client authorization */
```

Comment:
Note: We need to add man page entries for these new suboptions. No need to do it haxxpop, but we will need to do it before merging.